src/components/ProductsTable/ProductsTable.tsx

```diff
@@ -134,28 +114,29 @@ const ProductsTable: FunctionComponent<ProductsTableProps> = ({
       />
     );
   };
-
   const getPage = (products: Product[]) => {
     const first = (page - 1) * perPage;
     const last = first + perPage;
     return products.slice(first, last);
   };
-
   const filterMap = new Map<string, string>([
     ['active', 'Active'],
     ['expiringSoon', 'Expiring soon'],
     ['expired', 'Expired'],
     ['futureDated', 'Future dated']
   ]);
-
   const removeFilter = () => {
     setFilter('');
   };
-
-  const sortedProducts = sortProducts(data, activeSortIndex);
-  const searchedProducts = filterDataBySearchTerm(sortedProducts, searchValue);
+  const sortedProducts = data ? sortProducts(data, activeSortIndex) : [];
+  const filteredProducts = sortedProducts.filter((product) => {
+    if (!filter) return true;
+    return product.subscriptions?.some(
+      (subscription) => subscription.status.toLowerCase() === filter.toLowerCase()
+    );
+  });
```

Comment:
Suggestion: 

I believe this is unnecessary as the API call gets modified to be filtered on that end.

We should only get back products that have the correct status from the API, and therefore shouldn't need to filter here.